src/pip/_internal/commands/install.py

```diff
@@ -254,6 +261,26 @@ def run(self, options, args):
             target_temp_dir.create()
             install_options.append('--home=' + target_temp_dir.path)
 
+        if options.target_dir and options.scripts_target_dir:
+            options.scripts_target_dir = os.path.abspath(options.scripts_target_dir)
+            if (os.path.exists(options.target_dir) and not
+                    os.path.isdir(options.target_dir)):
+                raise CommandError(
+                    "Target path for scripts exists but is not a directory, "
+                    "will not continue."
+                )
+            if any(map(lambda x: x.startswith("--install-scripts"), install_options)):
+                raise CommandError(
+                    "Can not use '--scripts-target' and '--install-scripts' install "
+                    "command at the same time."
+                )
+            else:
+                install_options.append("--install-scripts="+options.scripts_target_dir)
```

Comment:
This should be reviewed for PEP 517. `install_options` is only relevant for legacy setuptools-based installs from source. It's probable that everything will work, as the code will build wheels for PEP 517 projects, and the wheel install code will be triggered below.

But the tests should definitely include installing a project from source, using both the legacy and the PEP 517 code paths, ensuring that both cases work.